### DISCUSSION TITLE
Investigate Codecov support for the GitHub Actions CI loop

### DISCUSSION THREAD

**Discussion by stebet** [rabbitmq/rabbitmq-dotnet-client#1529] [Ideas]
Since [Codecov](https://codecov.io) is free to use for Open Source Software it would be good to see if it makes sense to add that to the GitHub actions CI loops so we have better visibility into the Code Coverage of the integration tests for example, and to highlight areas for improvement.

As long as the licenses are compatible from a legal perspective of course.

Is this something you have looked at @lukebakken or @michaelklishin?

**Comment by lukebakken** [comment]
To be honest I have found code coverage to be something that people set up, then promptly ignore 😸

Related to this - there is a CodeQL workflow already in place that really I don't know what it accomplishes.

It looks like Codecov is already installed for github.com/rabbitmq, so I'll take a look at it.

**Reply by lukebakken** [reply]
https://github.com/rabbitmq/rabbitmq-dotnet-client/issues/1571

**Comment by michaelklishin** [comment]
I have found code coverage to be a pain more often than it's worth but we can try it, maybe this time will be different :)
